server/index.js

```diff
@@ -83,4 +83,35 @@ app.get('/itemsIncluded/:tierId', (req, res) => {
   })
 });
 
+app.get('/itemsByBundleId/:bundleId', (req, res) => {
+  const dataString = req.params;
+  return db.tiersRequest(dataString)
+  .then(tiersData => {
+    if (tiersData) {
+      return tiersData;
+    } else {
+      res.status(404).send('no such bundleId');
+    }
+  })
+  .then(data => {
+    let allItems = {};
+    for (let i = 0; i < data.length; i++) {
+      let currentTier = { tierId: data[i].dataValues.tierId };
+      return db.itemsRequest(currentTier)
+      .then(itemsData => {
+        if (itemsData) {
+          for (let j = 0; j < itemsData.length; j++) {
+            let currentItemId = itemsData[j].dataValues.itemId;
+            allItems[currentItemId] = currentItemId;
+          }
+        }
+        return allItems;
+      })
+    }
+  })
+  .then(itemsData => {
+    res.json(itemsData);
+  })
```

Comment:
Adding a catch here to send your 404 seems useful from a clarity perspective as well, so as not to chase errors through your chain.